src/global/core/components/animations/animation.ts

```diff
@@ -0,0 +1,1090 @@
+  let beforeStylesValue: { [property: string]: any } = {};
+  let afterAddClasses: string[] = [];
+  let afterRemoveClasses: string[] = [];
+  let afterStylesValue: { [property: string]: any } = {};
+  let numAnimationsRunning = 0;
+  let shouldForceLinearEasing = false;
+  let shouldForceSyncPlayback = false;
+  let cssAnimationsTimerFallback: any;
+  let forceDirectionValue: AnimationDirection | undefined;
+  let forceDurationValue: number | undefined;
+  let forceDelayValue: number | undefined;
+  let willComplete = true;
+  let finished = false;
+  let shouldCalculateNumAnimations = true;
+  let keyframeName: string | undefined;
+  // eslint-disable-next-line prefer-const
+  let ani: InterfaceAnimationInternal;
+
+  const id: string | undefined = animationId;
+  const onFinishCallbacks: InterfaceAnimationOnFinishCallback[] = [];
+  const onFinishOneTimeCallbacks: InterfaceAnimationOnFinishCallback[] = [];
+  const elements: HTMLElement[] = [];
+  const childAnimations: InterfaceAnimationInternal[] = [];
+  const stylesheets: HTMLElement[] = [];
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const _beforeAddReadFunctions: any[] = [];
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const _beforeAddWriteFunctions: any[] = [];
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const _afterAddReadFunctions: any[] = [];
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const _afterAddWriteFunctions: any[] = [];
+  const webAnimations: any[] = [];
+  const supportsAnimationEffect = (typeof (AnimationEffect as any) === 'function' || typeof (window as any).AnimationEffect === 'function');
+  const supportsWebAnimations = (typeof (Element as any) === 'function') && (typeof (Element as any).prototype.animate === 'function') && supportsAnimationEffect;
+  const ANIMATION_END_FALLBACK_PADDING_MS = 100;
+
+  const getWebAnimations = (): any[] => webAnimations;
+
+  /**
+   * Removes the animation's stylesheets
+   * from the DOM.
+   */
+  const cleanUpStyleSheets = (): void => {
+    stylesheets.forEach((stylesheet) => {
+
+      /**
+       * When sharing stylesheets, it's possible
+       * for another animation to have already
+       * cleaned up a particular stylesheet
+       */
+      if (stylesheet && stylesheet.parentNode) {
+        stylesheet.parentNode.removeChild(stylesheet);
+      }
+    });
+
+    stylesheets.length = 0;
+  };
+
+  /**
+   * Cancels any Web Animations and removes
+   * any animation properties from
+   * the animation's elements.
+   */
+  const cleanUpElements = (): void => {
+    if (supportsWebAnimations) {
+      webAnimations.forEach((animation) => {
+        animation.cancel();
+      });
+
+      webAnimations.length = 0;
+    } else {
+      const elementsArray = elements.slice();
+
+      raf(() => {
+        elementsArray.forEach((element) => {
+          removeStyleProperty(element, 'animation-name');
```

Comment:
Hi @kyubisation, 
about this, @simone-sabato and I were discussing whether or not to use the rest operator in the method signature.
We decided to go with

`removeStyleProperties = (element: HTMLElement, propertyNames: string[]): void`

because using

`removeStyleProperties = (element: HTMLElement, ...propertyNames: string[]): void`

if `propertyNames` is not passed, the compiler will see it as an empty array and will not complain;
obviously, within the method, `[].forEach(//do stuff)` will returns immediately, but it didn't make sense to us.
What do you think?

Reply:
Yeah, that's a good solution 👍